packages/test/service-load-test/src/loadTestDataStore.ts

```diff
@@ -222,26 +259,35 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
         const clientSendCount = config.testConfig.totalSendCount / config.testConfig.numClients;
         const opsPerCycle = config.testConfig.opRatePerMin * cycleMs / 60000;
         const opsGapMs = cycleMs / opsPerCycle;
-        while (dataModel.counter.value < clientSendCount && !this.disposed) {
-            if(dataModel.haveTaskLock()) {
-                dataModel.counter.increment(1);
-                if (dataModel.counter.value % opsPerCycle === 0) {
-                    dataModel.abandonTask();
-                    await wait(cycleMs);
+        try{
+            while (dataModel.counter.value < clientSendCount && !this.disposed) {
+                // this enables a quick ramp down. due to restart, some clients can lag
+                // leading to a slow ramp down. so if there are less than half the clients
+                // and it's partner is done, return true to complete the runner.
+                if(this.runtime.getAudience().getMembers().size < config.testConfig.numClients / 2
+                    && ((await dataModel.getPartnerCounter())?.value ?? 0) >= clientSendCount) {
+                    return true;
+                }
+
+                if(dataModel.haveTaskLock()) {
+                    dataModel.counter.increment(1);
+                    if (dataModel.counter.value % opsPerCycle === 0) {
+                        dataModel.abandonTask();
+                        // give our partner a half cycle to get the task
+                        await wait(cycleMs / 2);
+                    }else{
+                        // Random jitter of +- 50% of opWaitMs
+                        await wait(opsGapMs + opsGapMs * (Math.random() - 0.5));
+                    }
                 }else{
-                    // Random jitter of +- 50% of opWaitMs
-                    await wait(opsGapMs + opsGapMs * (Math.random() - 0.5));
+                    await dataModel.lockTask();
                 }
-            }else{
-                await dataModel.lockTask();
             }
+            return !this.runtime.disposed;
+        }finally{
+            clearTimeout(t);
+            dataModel.printStatus();
         }
-        dataModel.abandonTask();
```

Comment:
Doesn't this mean once one finishes, it's partner may be locked out? I know that in the end lone partners bail, but not until half the runners finish.

Reply:
taskmanager is fixed to remove clients that leave, so not explict abandom needed